src/global/signal_handler.cc

```diff
@@ -313,13 +313,33 @@ struct SignalHandler : public Thread {
 	    r = read(handlers[signum]->pipefd[0], &v, 1);
 	    if (r == 1) {
 	      siginfo_t * siginfo = &handlers[signum]->info_t;
-	      string task_name = get_name_by_pid(siginfo->si_pid);
-	      derr << "received  signal: " << sig_str(signum)
-		   << " from " << " PID: " << siginfo->si_pid
-		   << " task name: " << task_name
-		   << " UID: " << siginfo->si_uid
-		   << dendl;
-	      handlers[signum]->handler(signum);
+              ostringstream message;
+              message << "received  signal: " << sig_str(signum);
+              switch (siginfo->si_code) {
+                case SI_USER:
+                  message << " from " << get_name_by_pid(siginfo->si_pid);
+                  // If PID is undefined, it doesn't have a meaning to be displayed
+                  if (siginfo->si_pid) {
+                    message << " (PID: " << siginfo->si_pid << ")";
+                  } else {
+                    message << " ( Could be generated by pthread_kill(), raise(), abort(), alarm() )";
```

Comment:
it does not help from user's perspective. probably we can just print out the PID even it's zero.

Reply:
I put that as it took me some time to figure that this signal could be generated by the code itself and no only by a ctrl+c or explicit kill.

Reply:
well, how do we expect user to interpret this ?

> ( Could be generated by pthread_kill(), raise(), abort(), alarm() )

Reply:
It's not really for users but much more for advanced users or support.
I was in a support case where this code got triggered and we though for a while that the signal was sent by an external process as the error message was saying so.
It took sometime before considering the signal was not sent by an external tool but self-triggered by ceph itself.
If I had that message saying the signal could be triggered by ceph itself then I could have being focused on the real cause and not tracking my system & other processes.
That's why adding this information could save time for the next people facing this situation.